airflow/providers/ssh/hooks/ssh.py

```diff
@@ -148,7 +155,17 @@ def __init__(
                     self.pkey = self._pkey_from_private_key(private_key, passphrase=private_key_passphrase)
 
                 if "timeout" in extra_options:
-                    self.timeout = int(extra_options["timeout"], 10)
+                    warnings.warn(
+                        'Extra option `timeout` is deprecated.'
+                        'Please use `conn_timeout` instead.'
+                        'The old option `timeout` will be removed in a future version.',
+                        DeprecationWarning,
+                        stacklevel=2,
+                    )
+                    self.timeout = int(extra_options.get('timeout', 10))
```

Comment:
```suggestion
                    self.timeout = int(extra_options['timeout'])
```

We already know `"timeout"` is in `extra_options` (the `if` check above), this `get()` is not needed.